synapse/replication/tcp/handler.py

```diff
@@ -625,23 +625,6 @@ def on_REMOTE_SERVER_UP(
 
         self._notifier.notify_remote_server_up(cmd.data)
 
-        # We relay to all other connections to ensure every instance gets the
-        # notification.
-        #
-        # When configured to use redis we'll always only have one connection and
-        # so this is a no-op (all instances will have already received the same
-        # REMOTE_SERVER_UP command).
-        #
-        # For direct TCP connections this will relay to all other connections
-        # connected to us. When on master this will correctly fan out to all
-        # other direct TCP clients and on workers there'll only be the one
-        # connection to master.
-        #
-        # (The logic here should also be sound if we have a mix of Redis and
-        # direct TCP connections so long as there is only one traffic route
-        # between two instances, but that is not currently supported).
-        self.send_command(cmd, ignore_conn=conn)
```

Comment:
My reading of this comment is:

* On Redis we only have a single connection.
* When we get a `REMOTE_SERVER_UP` message we echo it back out to other connections, except ourselves.
* From that I deduce this is a no-op (which the comment says).

We can then remove this code, the code testing this situation and the additional parameter to `send_command`.